### ISSUE TITLE
chat does not work

### ISSUE THREAD

**Issue by emersonnpe** [Ylianst/MeshCentral#4884]
I use the server with OS Ubuntu 18.04 and I'm trying to use the chat on a machine with windows 10.

When opening the chat, it opens a tab in the browser and has the following message:

Waiting for the user.

When I copy the chat link in a new tab it lets me send the message, but it arrives on my machine and not on the destination machine. On the target machine does not open the browser.

Help me.

**Comment by LastStopITSolutions** [comment]
Im having the exact same issue.

Server is install on Ubuntu 20.04. 

Trying to create a chat with a windows machine and it just tells me waiting for other user. But nothing happens on the client machine. 

Any fix for this?

**Comment by si458** [comment]
I'll have a look in my free time for u @LastStopITSolutions 
So ur server os is ubuntu, the remote agent device is windows,
When u click chat, u get the chat window open in your web browser on your local device,
But no chat window opens on the remote windows device?
Are u running the latest version of meshcentral? 
Also the latest LTS nodejs version 18?

**Comment by LastStopITSolutions** [comment]
> I'll have a look in my free time for u @LastStopITSolutions So ur server os is ubuntu, the remote agent device is windows, When u click chat, u get the chat window open in your web browser on your local device, But no chat window opens on the remote windows device? Are u running the latest version of meshcentral? Also the latest LTS nodejs version 18?

Yes I am running LTS on everything. And correct, chat window opens on my device, but doesn't open on the clients PC.

**Comment by silversword411** [comment]
My default browser is brave. 
It opens a new tab in my browser when I start a chat session.

**Comment by LastStopITSolutions** [comment]
@silversword411 Does it open a corresponding chat on the clients browser as well?

**Comment by si458** [comment]
What's the remote OS? Is it Windows?

**Comment by LastStopITSolutions** [comment]
@si458 Yes. both the remote OS is windows and my OS is windows. MeshCentral is hosted on ubuntu.

**Comment by si458** [comment]
@LastStopITSolutions works perfectly fine here?
do u click the chat icon in the bottom right after you have clicked connect in remote tab?
whats the default browser set on the remote computer?
is the remote agent installed or do you run the assistant?

**Comment by LastStopITSolutions** [comment]
@si458 I click the chat button in the general tab.
![image](https://github.com/Ylianst/MeshCentral/assets/142823837/83978321-7a10-4d99-bd56-95813ad489cc)


And yes the agent is installed on the remote computer and default browser is set to chrome.

**Comment by si458** [comment]
from the looks of the meshcore code, it creates a task in the task schedular and then runs it,
it tries using the `win-tasks` module and if not possible then tries powershell
https://github.com/Ylianst/MeshCentral/blob/0a147a18d448d125992db929186d45613bda5223/agents/meshcore.js#L3600-L3638
im guessing the computer isnt in domain so might no be a permission issue
can u create a task manually with your user and run the below command?
`C:\\system32\\cmd.exe /C START https://www.google.com`

**Comment by LastStopITSolutions** [comment]
You want me to run that command on the remote machine or on my end via meshcentral?

**Comment by si458** [comment]
@LastStopITSolutions sorry please can u try running that command from the run prompt on the remote pc WIN+R
does it open up ur default browser to googles website?

**Comment by LastStopITSolutions** [comment]
Yes it does open a chrome browser.

**Comment by silversword411** [comment]
> @silversword411 Does it open a corresponding chat on the clients browser as well? 

Yes, I did a chat to myself it opened a 2nd browser tab

**Comment by si458** [comment]
@amoljagdalepucsd please can you try running the following command on a remote computer from INSIDE a powershell window?
`C:\Windows\system32\cmd.exe /C START https://www.google.com`
does a web browser open to google ok?

**Comment by si458** [comment]
ok i found the issue i think!
1. it tried to add a cmd task via scheduler
2. if step 1 fails it then falls back to add a cmd task via powershell BUT STILL USING THE TASK SCHEDULER!
3. step 2 will always fail because step 1 failed

we need yet another step that IF step 1+2+3 all fail, then just run the cmd,exe directly in powershell

**Comment by amoljagdalepucsd** [comment]
Thanks @si458 ,
I tried to open it manually inside power shell on remote computer and its opening browser.

**Comment by si458** [comment]
> Thanks @si458 , I tried to open it manually inside power shell on remote computer and its opening browser.

yes just what i throught! refer to above post! 

but to be honest i think u really need to fix your windows tasks scheduler, can you open the 'task scheduler' no problems on the remote computer without errors?

**Comment by si458** [comment]
@amoljagdalepucsd is ur remote device a normal device? Windows 10, 11?
as i think ive hit a problem, because the chat will only work with CONSOLE LOGGED IN devices, 
so it wont work if the remote device has a logged in user via RDP, only if they are physically logged on to it (as in on the screen)

**Comment by amoljagdalepucsd** [comment]
Hi @si458 ,
I checked this scenario on normal laptop devices [ **windows 10 , windows 11 Windows 11 Enterprise N** ] and **azure VM's** as well.
Henrique-Batista has put these details in our last discussion #5645.
I am not sure but I think If we provide all the required details [e. g user ,domain ,usersessionid ] from agent side same like console session then it should work.
I am checking it in more details from server and agent side as well.

Thank you @si458 for your time.

**Comment by amoljagdalepucsd** [comment]
Hey @si458 ,
I believe I encountered and successfully resolved the problem we were facing. Now, I'll verify that it's functioning correctly in all aspects without disrupting existing functionality.

**Comment by si458** [comment]
> Hey @si458 ,
> I believe I encountered and resolved the problem we were facing. Now, I'll verify that it's functioning correctly in all aspects without disrupting existing functionality.

That's great news! Can u share what was up? Was it the task scheduler? Or something else?

**Comment by si458** [comment]
@LastStopITSolutions do you still have this issue?
if so can you try opening up the windows task scheduler for me and see if you get any errors while trying to load it up?

**Comment by Maadhav** [comment]
I am also facing the same issue, please let me know how I can fix this

**Comment by amoljagdalepucsd** [comment]
@Maadhav , could you please share some details so that we can assist you in a better way

**Comment by Avessalom608** [comment]
@si458  I also have this problem. I also found that if I don't install the agent and only run it interactively (assuming the agent is not installed on the system), the problem goes away. Is there a solution at the moment?

**Comment by Avessalom608** [comment]
@amoljagdalepucsd Hello! I also have a problem at the moment.
help me please.
What details are you interested in?

Meshcentral version 2022, clients on Windows 10.
I noticed that if the agent is not installed, but only launched in interactive mode, then the problem does not occur.

**Comment by si458** [comment]
@emersonnpe @LastStopITSolutions @Avessalom608 @amoljagdalepucsd are you all this experiencing this issue?
do any of your computers that have this issue, in a domain environment? and login with a domain user and the chat isnt opening up?

**Comment by Lewis8265** [comment]
I too am facing this issue, the messenger just says always waiting for other user and nothing happens on the client.
